pytest_robotframework/_internal/pytest/robot_file_support.py

```diff
@@ -25,13 +28,32 @@
     from collections.abc import Iterable
     from os import PathLike
 
+    from _pytest._code.code import (
+        TerminalRepr,
+        _TracebackStyle,  # pyright: ignore[reportPrivateUsage]
+    )
+
 
 collected_robot_tests_key = StashKey[List[ModelTestCase]]()
 original_setup_key = StashKey[model.Keyword]()
 original_body_key = StashKey[Body]()
 original_teardown_key = StashKey[model.Keyword]()
 
 
+class StatusReporterButWorksProperly(StatusReporter):
+    @override
+    def _get_failure(
+        self, exc_type: type[BaseException], exc_value: BaseException, exc_tb: Never, context: Never
+    ):
+        result = super()._get_failure(exc_type, exc_value, exc_tb, context)  # pyright: ignore[reportUnknownMemberType]
+        if result:
+            result.error = exc_value  # pyright: ignore[reportAttributeAccessIssue]
+        return result
+
+
+librarykeywordrunner.StatusReporter = StatusReporterButWorksProperly
```

Comment:
Is this just `patch_method` with extra steps?

Reply:
Even if it is, I think it's worth it, status reporter discarding the exception has always been really frustrating.

Reply:
we don't want to patch every status reporter tho right, just a specific one?

Reply:
i guess just use `patch_method` then

Reply:
dome?